mitmproxy/proxy/protocol2/websocket.py

```diff
@@ -82,19 +92,29 @@ def relay_messages(self, event: events.Event) -> commands.TCommandGenerator:
                     yield commands.SendData(send_to, other.bytes_to_send())
 
                 elif isinstance(ws_event, wsevents.ConnectionClosed):
-                    yield commands.SendData(send_to, source.bytes_to_send())
-                    if isinstance(ws_event, wsevents.ConnectionFailed):
+                    other.close(ws_event.code, ws_event.reason)
+                    yield commands.SendData(send_to, other.bytes_to_send())
+                    # FIXME: Wait for other end to actually send the closing frame
```

Comment:
I think it's okay if we document this here and terminate both ends immediately. Makes the code considerably simpler. But yes, good catch. 👍